DataDog/datadog-operator
# [CECO-1763] Controller for generic custom resource

### What does this PR do?

Introduces a generic controller relying on Datadog Go client to handle CRUD operations on multiple resources. Initial support includes notebooks and synthetics (api/browser tests).
Examples:
```yaml
apiVersion: datadoghq.com/v1alpha1
kind: DatadogGenericResource
metadata:
  name: api-test
spec:
    type: synthetics_api_test
    jsonSpec: |-
        {
           [...]
        }
```

```yaml
apiVersion: datadoghq.com/v1alpha1
kind: DatadogGenericResource
metadata:
  name: notebook
spec:
    type: notebook
    jsonSpec: |-
        {
           [...]
        }
```
### Additional Notes

Next steps:

- Document fully-fledged examples
- Document onboarding strategy (e.g. export current resource from the API, create the associated YAML, deploy them)
- Document internally how to add new resources
- UI button to export to manifest
- Add option to start the controller in the Helm chart


### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

1. Include `-datadogGenericCREnabled=true` in `args` for the `manager` container to enable the controller, as well as `DD_API_KEY` and `DD_APP_KEY` in the envs
2. Deploy examples
3. Update their jsonSpec
4. Ensure the changes are reflected in the UI
5. Delete the resources
6. Ensure they are removed from the UI

### Checklist
